### PR TITLE
Update the II spec

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR removes old and outdated pieces of the II documentation and fixes the confusion between `hostname` and origin for the principal derivation section.

The API section was previously incomplete and has been expanded to match the actual API. The candid file has been restructured to be consistent and better organized.

The storage layout has changed a lot and is now documented in the relevant module itself: https://github.com/dfinity/internet-identity/blob/main/src/internet_identity/src/storage.rs

The user flows were removed since they do not describe a technical interface.

### CODE REVIEW

File: docs/ii-spec.mdx [hunk 1]
```diff
@@ -393,15 +436,42 @@ The method returns the expiration timestamp of the delegation. This is returned
...
-### The `get_delegation` query method
-
 For a certain amount of time after a call to `prepare_delegation`, a query call to `get_delegation` with the same arguments, plus the timestamp returned from `prepare_delegation`, actually fetches the delegation.
 
 Together with the `UserKey` returned by `prepare_delegation`, the result of this method is used by the Frontend to pass to the client application as per the [client authentication protocol](#client-authentication-protocol).
 
-**Authorization**: This request must be sent to the canister with `caller` that is the self-authenticating id derived from any of the public keys of devices associated with the user before this call.
+### Verifiable Credentials Protocol
+
+See here for the specification of the verifiable credentials protocol.
```
Comment (frederikrothenberger) [inline docs/ii-spec.mdx:448]: @jessiemongeon1: I'd like to link to the `vc-spec.md` file here. How do I do this correctly?

File: src/internet_identity/internet_identity.did [hunk 2]
```diff
@@ -507,7 +507,9 @@ type SignedIdAlias = record {
 };
 
 service : (opt InternetIdentityInit) -> {
-    init_salt: () -> ();
+    // Legacy identity management API
```
Comment (frederikrothenberger) [inline src/internet_identity/internet_identity.did:510]: This is simply reordering methods and adding headers.

File: docs/ii-spec.mdx [hunk 3]
```diff
@@ -293,34 +295,46 @@ In order to allow Internet Identity to read the path `/.well-known/ii-alternativ
 
 This section describes the interface that the backend canister provides.
 
-This interface is currently only used by its own frontend. This tight coupling means that this interface may change, even in incompatible ways. We therefore do not have to apply Candid best practices for backward-compatibility (such as using records for arguments and results).
+Note that the interface is split into 4 categories:
```
Comment (lmuntaner) [inline docs/ii-spec.mdx:298]: Why did you remove the warning paragraph of breaking changes?

> This interface is currently only used by its own frontend. This tight coupling means that this interface may change, even in incompatible ways. We therefore do not have to apply Candid best practices for backward-compatibility (such as using records for arguments and results)

File: docs/ii-spec.mdx [hunk 4]
```diff
@@ -293,34 +295,46 @@ In order to allow Internet Identity to read the path `/.well-known/ii-alternativ
...
+    * these are methods related to initialisation of II itself and integration with its archive
 
 The summary is given by the Candid interface:
 
 <CodeBlock language="candid">{IICandidInterface}</CodeBlock>
 
-The `init_salt` method is mostly internal, see [Salt](#salt).
-
-### The `register` and `create_challenge` methods
+### Identity Management (legacy API)
```
Comment (lmuntaner) [inline docs/ii-spec.mdx:315]: How about documenting the legacy in the candid interface and adding a link to it above instead of repeating the duplicating the data?

Did you discuss with someone whether we still want the endpoint descriptions in this docs?

**Review by lmuntaner — APPROVED** [review]
A couple of comments, but it looks very good. I would even remove more docs.

File: docs/ii-spec.mdx (hunk 3 above)
Comment (frederikrothenberger) [reply docs/ii-spec.mdx:298]: Well, because it is not true. There are at least 2 other products that use the II back-end in some capacity.

File: docs/ii-spec.mdx (hunk 4 above)
Comment (frederikrothenberger) [reply docs/ii-spec.mdx:315]: I did it this way because it is then aligned with how the interface spec does it. See e.g. the management canister API spec: https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-management-canister

I briefly discussed with @jessiemongeon1. But I'm open to change it and just have the candid file.
What do you think @jessiemongeon1?

File: docs/ii-spec.mdx (hunk 3 above)
Comment (lmuntaner) [reply docs/ii-spec.mdx:298]: Oh, I didn't know! Ok, in that case, we do need to keep backwards compatibility. Good to know!

File: docs/ii-spec.mdx (hunk 4 above)
Comment (jessiemongeon1) [inline docs/ii-spec.mdx:315]:
```suggestion
### Identity management (legacy API)
```

Comment (jessiemongeon1) [reply docs/ii-spec.mdx:315]: I think aligning with the Interface Spec is the best approach for consistency.

File: docs/ii-spec.mdx [hunk 5]
```diff
@@ -329,59 +343,88 @@ It is allowed to remove the last key, to completely disable a user. The canister
...
+Fetches all data associated with an anchor including registration mode and tentatively registered devices.
 
-### The `get_anchor_info` method
+#### The `get_principal` query method
+**Authorization**: This request must be sent to the canister with `caller` that is the self-authenticating id derived from any of the public keys of devices associated with the user before this call.
 
-Fetches all data associated with an anchor including registration mode and tentatively registered devices.
+Fetches the principal for a given user and front end.
 
+### Identity Management (API V2 only)
```
Comment (jessiemongeon1) [inline docs/ii-spec.mdx:409]:
```suggestion
### Identity management (API V2 only)
```

File: docs/ii-spec.mdx (hunk 1 above)
Comment (jessiemongeon1) [reply docs/ii-spec.mdx:448]:
```
See [here](vc-spec.md) for the specification of the verifiable credentials protocol.
```

File: docs/ii-spec.mdx [hunk 6]
```diff
@@ -393,15 +436,42 @@ The method returns the expiration timestamp of the delegation. This is returned
...
 For a certain amount of time after a call to `prepare_delegation`, a query call to `get_delegation` with the same arguments, plus the timestamp returned from `prepare_delegation`, actually fetches the delegation.
 
 Together with the `UserKey` returned by `prepare_delegation`, the result of this method is used by the Frontend to pass to the client application as per the [client authentication protocol](#client-authentication-protocol).
 
-**Authorization**: This request must be sent to the canister with `caller` that is the self-authenticating id derived from any of the public keys of devices associated with the user before this call.
+### Verifiable Credentials Protocol
+
+See here for the specification of the verifiable credentials protocol.
+
+### HTTP Gateway Protocol
```
Comment (jessiemongeon1) [inline docs/ii-spec.mdx:450]:
```suggestion
### HTTP gateway protocol
```

File: docs/ii-spec.mdx [hunk 7]
```diff
@@ -393,15 +436,42 @@ The method returns the expiration timestamp of the delegation. This is returned
...
+**Authorization**: Can only be called by anyone.
+
+The `deploy_archive` method deploys the supplied wasm to the archive canister, given it matches the configured archive wasm hash.
+
+#### The `fetch_entries` and `acknowledge_entries` methods
+**Authorization**: Can only be called by the archive canister.
+
+API for the archive canister to fetch archive entries from the Internet Identity canister and to trigger pruning of archived entries within Internet Identity.
 
 ## The Internet Identity Service backend internals
```
Comment (jessiemongeon1) [inline docs/ii-spec.mdx:476]:
```suggestion
## The Internet Identity service backend internals
```

**Review by jessiemongeon1 — APPROVED** [review]
